docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,45 @@
-# Cluster
+# Cluster Controller
+
+The cluster controller is responsible for keeping the cluster object in sync with the infrastructure cluster object.
+
+Its main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object.
+* Deleting all related objects in the correct order.
+* Keeping the Cluster object up to date with external objects.
```

Comment:
This might need a little bit more definition, what's an external object and what are we syncing?